### DISCUSSION TITLE
Delete attribute from SQLAlchemy model object

### DISCUSSION THREAD

**Discussion by ahmadeposfirst** [sqlalchemy/sqlalchemy#6614] [Usage Questions] [answered]
Hi there,

First of all thanks for amazing work on SQLAlchemy.

I'm trying to delete an attribute from model object that is created on the go, but there seems to be some problem doing so. So, far I've searched through every place including SQLAlchemy documentation and some of its code to find a potential fix but unable to find one.

Below code works fine on python classes but not with a class inherited from ```declarative_base```
```
    for row in data:
        model_obj = DBEngine.models.User()
        [setattr(model_obj, key, value) for key, value in row.items()]

        # below line doesn't work as expected, instead of deleting the 
        # attribute it just sets the value of attribute to None
        delattr(model_obj, 'localedit')

        session.add(model_obj)
```

User model
```
Base = declarative_base()

class User(Base):
    __tablename__ = 'user'

    id        = Column('id', Integer, primary_key=True)
    localedit = Column('localedit', String,  default="0000-00-00 00:00:00")
```

I've also tried deleting the attribute using ```del``` but it's doing the same job as ```delattr```, I guess both ```del``` and ```delattr``` calling the same code under the hood.

I'm totally stumped on this problem and couldn't come up with anything. 
Any help would be appreciated, Thanks.

**Comment by zzzeek** [comment]
hey there -

so this is expected, SQLAlchemy ORM mapped objects don't support this particular state for an attribute, that is, attribute doesn't exist and would raise AttributeError.   for an ORM mapped class, a mapped attribute always defaults to None and/or empty collection.  there's a little bit of an introduction to this here: https://docs.sqlalchemy.org/en/14/tutorial/orm_data_manipulation.html#instances-of-classes-represent-rows

**Reply by ahmadeposfirst** [reply]
@zzzeek thanks for your response. 
I'll try to explain what I'm ought to be doing. I have two same tables on different places, one is local and it has one field that is not present on the second database table. Is there any way I can achieve insertion in the second database using the same model without raising Unknow Column error.

**Reply by zzzeek** [reply]
you can either make a single mapping/Table that omits that one column which we assume you don't care about, or you have to make two separate mapping/Tables for each database.

**Comment by ahmadeposfirst** [comment, marked as answer]
Thanks for your consideration, I have solved it by changing the default value of ```localedit``` to ```FetchedValue()```

For this particular problem, one can define column as
```
localedit = Column('localedit', String,  FetchedValue())
```

FetchedValue is used when the database is configured to provide some automatic default for a column. So, in this case I only want to ignore the column and this would work like a charm. Updated model would look something like:

```
Base = declarative_base()

class User(Base):
    __tablename__ = 'user'

    id        = Column('id', Integer, primary_key=True)
    localedit = Column('localedit', String,  FetchedValue())
```
